src/openrct2/ride/Vehicle.cpp

```diff
@@ -5560,15 +5560,13 @@ void Vehicle::UpdateVelocity()
     }
     if (HasFlag(VehicleFlags::StoppedOnHoldingBrake))
     {
-        vertical_drop_countdown--;
-        if (vertical_drop_countdown == -70)
+        if (vertical_drop_countdown > 0)
         {
             ClearFlag(VehicleFlags::StoppedOnHoldingBrake);
-        }
-        if (vertical_drop_countdown >= 0)
-        {
+
             nextVelocity = 0;
             acceleration = 0;
```

Comment:
Previously, `nextVelocity` and `acceleration` were set to 0 if `vertical_drop_countdown == 0`, but now it doesn’t. Could you elaborate on this? I did read the PR description, but this wasn’t clear to me.

Reply:
The existing code checks if vertical_drop_countdown is greater than or equal to zero but it also decrements the counter before doing the check, so the velocity and acceleration are set to zero only if UpdateVelocity is called with vertical_drop_countdown strictly greater than zero.

In my version, I don't decrement the counter until after the check so I changed the comparison to a strict inequality to ensure the same behaviour, otherwise the train would wait an extra frame. In my version it stops counting down when the timer reaches zero, whereas the original will release the train but keep decrementing vertical_drop_timer until it hits -70 and clear then the flag. My version clears the flag upon leaving the track piece.

Reply:
Ah right, I see.